src/boot/efi/boot.c

```diff
@@ -2292,6 +2328,131 @@ static void config_entry_add_linux(
         }
 }
 
+static EFI_STATUS secure_boot_enroll(EFI_FILE *root_dir, Config *config, ConfigEntry *entry) {
```

Comment:
i think this function is long enough to deserve its own .c/.h file pair

Reply:
The only reason I haven't done it is its reliance on Config and ConfigEntry. Should I still split it off?

Reply:
yeah, add boot.h and move definition of Config/ConfigEntry there. Sooner or later we really have to do that anyway